File: Programmers/Java/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        int ans = 0;
        int size = numbers.length;

        for(int i = 0; i < 1 << size; i++){
            int sum = 0;
            for(int j = 0; j < size; j++){
                if(((i >> j) & 1) == 1) sum += numbers[j];
                else sum -= numbers[j];
            }
            if(sum == target) ans++;
        }
        return ans;
    }
}

File: Programmers/Python/타겟 넘버.py
def go(numbers,target,size,idx,tot):
    if(size == idx):
        if(target == tot):
            return 1
        else:
            return 0

    return go(numbers,target,size,idx+1,tot+numbers[idx])+go(numbers,target,size,idx+1,tot-numbers[idx])

def solution(numbers, target):
    return go(numbers,target,len(numbers),0,0)
